docs/databricks-monitoring/databricks-monitoring.md

```diff
@@ -0,0 +1,182 @@
+8. Click the "create cluster" button to create the cluster. Next, click on the "start" button to start the cluster.
+
+## Use the monitoring library in your code
+
+The Azare Databricks monitoring library supports the following types of metrics and message logging:
+
+### Apache Spark level events
+
+To send Apache Spark level event from Azure Databricks to your Azure Log Analytics workspace, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners** project.
+2. Once this is successfully deployed, metrics flow to your Azure Log Analytics workspace. You do not need to make any changes to your application code.
+
+### Apache Spark structured streaming sink for Azure Log Analytics
+
+To deploy the Apache Spark structured streaming sink for Azure Log Analytics in your Azure Databricks workspace, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-loganalytics-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners-loganalytics** project.
+2. Once this is successfully deployed, metrics flow to your Azure Log Analytics workspace. You do not need to make any changes to your application code.
+
+### Azure Databricks application metrics using Dropwizard
+
+To send application [metrics](https://spark.apache.org/docs/latest/monitoring.html#metrics) from your Azure Databricks application code to your Azure Log Analytics workspace using Dropwizard, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-loganalytics-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners-loganalytics** project.
+2. Create any [Dropwizard gauges or counters](https://metrics.dropwizard.io/4.0.0/manual/core.html) in your application code. 
+
+The code library includes a sample application that demonstrates how to implement custom DropWizard metrics. The **StreamingQueryListenerSampleJob** class creates an instance of the **UserMetricsSystem** class. A discussion of these classes is beyond the scope of this document, however, these can be used as the basis for your own custom metrics system.  
+
+### Azure Databrick log4j Appender
+
+To send your Azure Databricks application logs to Azure Log Analytics using the [log4j appender](https://logging.apache.org/log4j/2.x/manual/appenders.html) in the library, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-loganalytics-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners-loganalytics** project.
+2. In your application code, include the **spark-listeners-loganalytics** project, and `import com.microsoft.pnp.logging.Log4jconfiguration` to your application code.
+3. Create a **log4j.properties** file for your application. In addition to any properties that you specify, you must include the following and substitute your application package name and log level where indicated:
```

Comment:
This is one way to configure log4j on a Spark cluster.  The requirement to use the package name and log level is not strictly required.  If it is omitted, all Spark logs will flow to Azure Log Analytics.  That is likely not what the user wants, in this scenario.

However, if the user want to log all Spark logs to Azure Log Analytics, they can edit the cluster init script to modify the /databricks/spark/conf/log4j.properties file and add their log4j properties there.